code/controllers/subsystem/evacuation.dm

```diff
@@ -0,0 +1,227 @@
+	runlevels = RUNLEVEL_GAME
+	/// Controllers that handle the evacuation of the station
+	var/list/datum/evacuation_controller/controllers = list()
+	/// A list of things blocking evacuation
+	var/list/evacuation_blockers = list()
+	/// Whether you can cancel evacuation. Used by automatic evacuation
+	var/cancel_blocked = FALSE
+
+/datum/controller/subsystem/evacuation/Initialize(start_timeofday)
+	for(var/path in SSmapping.config.evacuation_controllers)
+		var/datum/evacuation_controller/controller = new path
+		controllers[controller.id] = controller
+	return ..()
+
+/datum/controller/subsystem/evacuation/fire(resumed)
+	if(!SSticker.HasRoundStarted() || length(evacuation_blockers))
+		return
+
+	var/threshold = CONFIG_GET(number/evacuation_autocall_threshold)
+	if(!threshold)
+		return
+
+	var/alive = 0
+	for(var/mob/M as anything in GLOB.player_list)
+		if(M.stat != DEAD)
+			++alive
+
+	var/total = length(GLOB.joined_player_list)
+	if(total <= 0)
+		return
+
+	if(alive / total > threshold)
+		return
+
+	message_admins(CREW_DEATH_MESSAGE)
+	log_evacuation("[CREW_DEATH_MESSAGE] Alive: [alive], Roundstart: [total], Threshold: [threshold]")
+	priority_announce("Catastrophic casualties detected: crisis evacuation protocols activated - blocking recall signals across all channels.")
+	trigger_auto_evac(EVACUATION_REASON_CREW_DEATH)
+
+/datum/controller/subsystem/evacuation/proc/trigger_auto_evac(reason)
+	cancel_blocked = TRUE
+	for(var/identifier in controllers)
+		if(controllers[identifier].state != EVACUATION_STATE_IDLE)
+			return
+		if(controllers[identifier].start_automatic_evacuation(reason))
+			return
+
+/datum/controller/subsystem/evacuation/proc/can_evac(mob/caller, controller_id)
+	var/datum/evacuation_controller/controller = controllers[controller_id]
+	if(!controller)
+		return "Error 500. Please contact your system administrator."
+	for(var/identifier in controllers)
+		if(controllers[identifier].state >= EVACUATION_STATE_AWAITING)
+			return "Evacuation is already in progress."
+	return controller.can_evac(caller)
+
+/datum/controller/subsystem/evacuation/proc/request_evacuation(mob/caller, reason, controller_id, admin = FALSE)
+	var/datum/evacuation_controller/controller = controllers[controller_id]
+	if(!controller)
+		return
+	for(var/identifier in controllers)
+		if(controllers[identifier].state >= EVACUATION_STATE_AWAITING)
+			to_chat(caller, "Evacuation is already in progress.")
+			return
+	return controller.trigger_evacuation(caller, reason, admin)
+
+/datum/controller/subsystem/evacuation/proc/can_cancel(mob/caller, controller_id)
+	var/datum/evacuation_controller/controller = controllers[controller_id]
+	if(!controller)
+		return FALSE
+	return controller.can_cancel(caller)
+
+/datum/controller/subsystem/evacuation/proc/request_cancel(mob/caller, controller_id)
+	var/datum/evacuation_controller/controller = controllers[controller_id]
+	if(!controller)
+		return
+	controller.trigger_cancel_evacuation(caller)
```

Comment:
If you're trying to cancel a nonexistent controller, something's likely gone severely wrong. Consider making these throw a stack or outright crash?